K8s install page refresh

jpmartinspt opened the pull request:
## Done

- K8s install page refresh

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/kubernetes/install
- Check the copy matches what is on the copy doc

## Issue / Card

Fixes #https://github.com/canonical-web-and-design/web-squad/issues/3984

webteam-app commented:
Demo starting at https://ubuntu-com-9656.demos.haus

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-      </div>
+    <table class="p-table" style="margin-top: 1rem;">
+      <thead>
+        <tr style="background: #E5E5E5;">
```
Why change the base table styling?

jpmartinspt replied:
It's removed know, I was experimenting and forgot to remove the style.

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-          <li class="p-list__item is-ticked"><strong>Small edge clouds:</strong> create simple Kubernetes clusters for your edge clouds</li>
-        </ul>
-      </div>
+    <table class="p-table" style="margin-top: 1rem;">
```
Could you try and add mobile card to make it easier to read on small screens.

jpmartinspt replied:
Added now, let me know what you think.

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+      </thead>
+      <tbody>
+        <tr>
+          <td><a href="https://microk8s.io">MicroK8s</a></td>
```
External link

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+    </table>
+    <p>
+      <a href="/kubernetes/contact-us?product=kubernetes-install">Looking for help running Kubernetes? Get in touch
+        &nbsp;&rsaquo;</a>
```
The &nbsp; should be touching the last character of the link otherwise it will wrap. Defeating the purpose of the nbsp

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+        upstream Kubernetes services and their dependencies. With support for x86 and ARM64, MicroK8s runs from local
+        workstations to the edge and IoT appliances.</p>
+      <p>
+        <a href="https://microk8s.io/">Learn more about microk8s &nbsp;&rsaquo;</a>
```
External link

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+      <p class="p-heading--four" class="p-heading--five p-tab__title">Linux</p>
+    </a>
+
+    <a id="tab-windwos" href="#tab-windows__content" class="col-small-2 col-medium-2 col-4 p-card p-tab u-align--center u-vertically-center js-tab" role="tab" aria-selected="false">
```
Typo "windwos"

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+        bare-metal to the cloud. Build your Kubernetes cloud from the ground up, integrate it with your favorite tools,
+        and create multi-cloud topologies.</p>
+      <p>
+        <a href="/kubernetes/features">Learn more about Charmed Kubernetes &nbsp;&rsaquo;</a>
```
No space before the nbsp

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
+      <pre class="p-code-snippet__block--icon"><code>microk8s kubectl get no</code></pre>
+    </div>
+    <p>
+      <a href="https://microk8s.io/docs/clustering">Learn more about MicroK8s clustering &nbsp;&rsaquo;</a>
```
Space before nbsp

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-          <p><a class="p-link--external" href="https://jaas.ai/docs/clouds">Find out more about Clouds in Juju</a></p>
+      </div>
+      <p>If you want to know more about our enterprise multi-cloud Kubernetes read the <a
+          href="/kubernetes/docs">Charmed Kubernetes documentation &nbsp;&rsaquo;</a></p>
```
Space before nbsp

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-          </div>
+      </div>
+      <p>Are you a developer looking to quickly run local K8s clusters or an enterprise looking for K8s at the edge?
+        Read the <a href="https://microk8s.io/docs">MicroK8s documentation &nbsp;&rsaquo;</a></p>
```
External link

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-          </div>
+      </div>
+      <p>Curious to learn about our Charmed Operators and how they automate DevOps and application lifecycle tasks?
+        Browse through the <a href="https://jaas.ai/docs">Charmed OLM documentation &nbsp;&rsaquo;</a></p>
```
External link

anthonydillon commented on templates/kubernetes/install.html:
```diff
@@ -2,338 +2,634 @@
-        </li>
-      </ol>
+      </div>
+      <a class="p-link--external p-link--inverted" href="/engage/kubernetes-deployment-enterprise-whitepaper">Five strategies to accelerate Kubernetes deployment in the enterprise</a>
```
Not an external link

anthonydillon commented on templates/kubernetes/_install-windows.html:
```diff
@@ -0,0 +1,107 @@
+      Download the installer for Windows
+    </h3>
+    <div class="p-stepped-list__content">
+      <p><a href="https://github.com/ubuntu/microk8s/releases/download/installer-v2.0.0/microk8s-installer.exe">Download MicroK8s for Windows</a></p>
```
External link

anthonydillon commented on templates/kubernetes/_install-windows.html:
```diff
@@ -0,0 +1,107 @@
+      </div>
+      <p>If you mainly use MicroK8s you can run the native Windows version of kubectl on your command-line.</p>
+      <p>
+        <a href="https://microk8s.io/docs/working-with-kubectl#heading--kubectl-windows">Learn how &nbsp;&rsaquo;</a>
```
Space before nbsp

anthonydillon commented on templates/kubernetes/_install-windows.html:
```diff
@@ -0,0 +1,107 @@
+    <div class="p-stepped-list__content">
+      <p>Kubernetes is a collection of system services that talk to each other all the time. If you don’t need them running in the background then you will save battery by stopping them. microk8s start and microk8s stop will do the work for you.</p>
+      <p>
+        <a href="https://microk8s.io/docs">Read the docs to learn more &nbsp;&rsaquo;</a>
```
External link

anthonydillon reviewed (changes requested):
Looking good, few issues with links and there are a few validation issues: https://validator.w3.org/nu/?doc=https%3A%2F%2Fubuntu-com-9656.demos.haus%2Fkubernetes%2Finstall

elioqoshi commented:
@jpmartinspt The headings were wrong in the copydoc so I fixed their hierarchy now. Could you check to ensure they are synced?

I am also not sure regarding the consistency for the first install steps. There is a generic Linux logo with a label while Windows and MacOS have no labels (because of their logo wordmark I assume). Either we remove the labels or have labels for every card (I'd suggest latter). Also, the multi-node install steps platform cards have a higher height than the single-node cards. Could they be reduced to be the same? The Linux logo might need to be smaller then

elioqoshi commented:
Also maybe this is a stretch but the install instructions are hard to disttinguish visually frorm everything else, maybe nest it within a different background with an arrow pointing to the card? Similar to this:

![image](https://user-images.githubusercontent.com/5436686/116566971-7fdd3200-a907-11eb-9fdb-cf7e2f2c93ad.png)

If the arrow is too complicated, we can leave it out as well.

cc @anasereijo of course if that makes sense.

jpmartinspt commented:
@anthonydillon thanks for the review, addressed your comments now.
@elioqoshi made the backgrounds change on instructions and reviewed heading sizes.

anthonydillon commented:
Thanks @jpmartinspt I'll review again today. Maybe after the design review.

anasereijo commented:
thanks, @jpmartinspt this is looking much better.

- [ ] Just wondering if it's possible to have the arrow when selecting one of the cards? we might need to add a border to it too. something like what we have on https://ubuntu.com/advantage/subscribe

<img width="1172" alt="Screenshot 2021-04-30 at 09 21 16" src="https://user-images.githubusercontent.com/36884067/116668838-c2e5e680-a995-11eb-8aad-1bd69abd9964.png">


- [ ] @elioqoshi from a UX perspective do you think it's better to have the first card selected/opened when we load the page? I think it's nicer when there are no cards selected, also easier for the user to quickly see the options: Single node or Multi node.

- [ ] Can we remove the labels/names from these cards? I think the logos are enough:

<img width="1105" alt="Screenshot 2021-04-30 at 09 28 24" src="https://user-images.githubusercontent.com/36884067/116669465-823a9d00-a996-11eb-9a61-df6124e523a2.png">

- [ ] Resources section: please match the headings and spacing with this example: https://ubuntu.com/0_docs/content_strips

elioqoshi commented:
> @elioqoshi from a UX perspective do you think it's better to have the first card selected/opened when we load the page? I think it's nicer when there are no cards selected, also easier for the user to quickly see the options: Single node or Multi node.

Yeah it makes sense to have none selected on default.

> Can we remove the labels/names from these cards? I think the logos are enough:

We should remove all labels then, also in multi-node install steps.

anasereijo commented:
@jpmartinspt thanks for making the changes! looks good 👍 
@anthonydillon ready for a code review

pmahnke reviewed (approved):
+1
